src/PathDoesNotExistException.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Amp\File;
+
+use Exception;
+
+class PathDoesNotExistException extends FileSystemException
+{
+    public function __construct(string $path)
+    {
+        parent::__construct(sprintf('Specified path "%s" does not exist', $path));
+    }
+}
```

Comment:
I recommend also adding a getter for the path. It might help custom exception handling.